src/mds/MDSRank.cc

```diff
@@ -3455,17 +3455,20 @@ void MDSRankDispatcher::handle_osd_map()
 
   purge_queue.update_op_limit(*mdsmap);
 
-  std::set<entity_addr_t> newly_blocklisted;
-  objecter->consume_blocklist_events(&newly_blocklisted);
-  auto epoch = objecter->with_osdmap([](const OSDMap &o){return o.get_epoch();});
-  dout(4) << "handle_osd_map epoch " << epoch << ", "
-          << newly_blocklisted.size() << " new blocklist entries" << dendl;
-  auto victims = server->apply_blocklist(newly_blocklisted);
-  if (victims) {
-    set_osd_epoch_barrier(epoch);
+  // its ok if reaplay state is reached via standby-replay, the
+  // reconnect state will journal blocklisted clients.
+  if (!is_replay()) {
+    std::set<entity_addr_t> newly_blocklisted;
+    objecter->consume_blocklist_events(&newly_blocklisted);
+    auto epoch = objecter->with_osdmap([](const OSDMap &o){return o.get_epoch();});
+    dout(4) << "handle_osd_map epoch " << epoch << ", "
+            << newly_blocklisted.size() << " new blocklist entries" << dendl;
+    auto victims = server->apply_blocklist(newly_blocklisted);
+    if (victims) {
+      set_osd_epoch_barrier(epoch);
+    }
```

Comment:
This code is also repeated in `reconnect_start`. Let's create a helper method while fixing this?